cmake/modules/AddSwift.cmake

```diff
@@ -1849,8 +1849,13 @@ function(add_swift_target_library name)
 
       foreach(lib ${SWIFTLIB_PRIVATE_LINK_LIBRARIES})
         if("${lib}" STREQUAL "ICU_UC")
-          list(APPEND swiftlib_private_link_libraries_targets
-               "${SWIFT_${sdk}_${arch}_ICU_UC}")
+          if("${sdk}" STREQUAL "ANDROID" OR "${SWIFT_PATH_TO_LIBICU_BUILD}" STREQUAL "")
+            list(APPEND swiftlib_private_link_libraries_targets
+                 "${SWIFT_${sdk}_${arch}_ICU_UC}")
+          else()
+            list(APPEND swiftlib_private_link_libraries_targets -licuuc -licudata)
```

Comment:
Probably this will have to change if libicu gets more integrated into the build system, but for the moment, this change only moves code that was in `stdlib/public/core/CMakeLists.txt` into here. The code has been moved as similar as possible, and that’s why I kept using `-licuuc` and `-licudata` here.